apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,61 +1,46 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  TemplateRef,
+  input,
+  output,
+} from '@angular/core';
+import { ListItemRefDirective } from '../list-item/list-item-ref.directive';
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
   template: `
-    <div
-      class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
-      [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
-
-      <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
-      </section>
-
-      <button
-        class="rounded-sm border border-blue-500 bg-blue-300 p-2"
-        (click)="addNewItem()">
-        Add
-      </button>
-    </div>
+    <ng-content select="img"></ng-content>
+    <section>
+      @for (item of list(); track item.id) {
+        <ng-template
+          [ngTemplateOutlet]="listItemTemplate"
+          [ngTemplateOutletContext]="{ $implicit: item }"></ng-template>
+      } @empty {
+        <div>There is no item</div>
+      }
+    </section>
+    <button
+      class="rounded-sm border border-blue-500 bg-blue-300 p-2"
+      (click)="addItem.emit()">
+      Add
+    </button>
   `,
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  host: {
+    class: 'flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4',
+  },
+  imports: [NgIf, NgFor, ListItemComponent, NgTemplateOutlet],
```

Comment:
no need for ngif and ngfor

Reply:
Sorry, I forgot to delete them from imports array. I will fix it.